src/detector/implementations/aruco_marker_implementation.py

```diff
@@ -0,0 +1,280 @@
+                        f"{self._marker_parameters.corner_refinement_method}.")
+        corner_refinement_method_text: CornerRefinementMethod = \
+            CORNER_REFINEMENT_METHOD_DICTIONARY_INT_TO_TEXT[self._marker_parameters.cornerRefinementMethod]
+        params: DetectionParameters = DetectionParameters(
+            adaptive_thresh_constant=self._marker_parameters.adaptiveThreshConstant,
+            adaptive_thresh_win_size_max=self._marker_parameters.adaptiveThreshWinSizeMax,
+            adaptive_thresh_win_size_min=self._marker_parameters.adaptiveThreshWinSizeMin,
+            adaptive_thresh_win_size_step=self._marker_parameters.adaptiveThreshWinSizeStep,
+            april_tag_critical_rad=self._marker_parameters.aprilTagCriticalRad,
+            april_tag_deglitch=self._marker_parameters.aprilTagDeglitch,
+            april_tag_max_line_fit_mse=self._marker_parameters.aprilTagMaxLineFitMse,
+            april_tag_max_nmaxima=self._marker_parameters.aprilTagMaxNmaxima,
+            april_tag_min_cluster_pixels=self._marker_parameters.aprilTagMinClusterPixels,
+            april_tag_min_white_black_diff=self._marker_parameters.aprilTagMinWhiteBlackDiff,
+            april_tag_quad_decimate=self._marker_parameters.aprilTagQuadDecimate,
+            april_tag_quad_sigma=self._marker_parameters.aprilTagQuadSigma,
+            corner_refinement_max_iterations=self._marker_parameters.cornerRefinementMaxIterations,
+            corner_refinement_method=corner_refinement_method_text,
+            corner_refinement_min_accuracy=self._marker_parameters.cornerRefinementMinAccuracy,
+            corner_refinement_win_size=self._marker_parameters.cornerRefinementWinSize,
+            detect_inverted_marker=self._marker_parameters.detectInvertedMarker,
+            error_correction_rate=self._marker_parameters.errorCorrectionRate,
+            marker_border_bits=self._marker_parameters.markerBorderBits,
+            max_erroneous_bits_in_border_rate=self._marker_parameters.maxErroneousBitsInBorderRate,
+            max_marker_perimeter_rate=self._marker_parameters.maxMarkerPerimeterRate,
+            min_corner_distance_rate=self._marker_parameters.minCornerDistanceRate,
+            min_distance_to_border=self._marker_parameters.minDistanceToBorder,
+            min_marker_distance_rate=self._marker_parameters.minMarkerDistanceRate,
+            min_marker_perimeter_rate=self._marker_parameters.minMarkerPerimeterRate,
+            min_otsu_std_dev=self._marker_parameters.minOtsuStdDev,
+            perspective_remove_ignored_margin_per_cell=self._marker_parameters.perspectiveRemoveIgnoredMarginPerCell,
+            perspective_remove_pixel_per_cell=self._marker_parameters.perspectiveRemovePixelPerCell,
+            polygonal_approx_accuracy_rate=self._marker_parameters.polygonalApproxAccuracyRate)
+        return GetDetectionParametersResponse(parameters=params)
+
+    # Detector?
```

Comment:
We'll need some kind of function that returns the 2D points. You could have the `internal_update_markers` function do that. But in that case, please change its name (and type hints) to reflect that it does that.